sdk/src/azure/core/az_http_request.c

```diff
@@ -107,11 +171,45 @@ az_http_request_set_query_parameter(_az_http_request* ref_request, az_span name,
   // name or value can't be empty
   _az_PRECONDITION(az_span_size(name) > 0 && az_span_size(value) > 0);
 
-  int32_t required_length = az_span_size(name) + az_span_size(value) + 2;
-
+  // check if query parameter is already in url
+  az_span prexisting_query_parameter_value
+      = _az_http_request_find_query_parameter(ref_request, name);
   az_span url_remainder
       = az_span_slice_to_end(ref_request->_internal.url, ref_request->_internal.url_length);
 
+  if (az_span_ptr(prexisting_query_parameter_value) != NULL)
```

Comment:
nit: Instead of checking the ptr is null, check that size == 0. Reduces the calls to az_span_ptr that we'd need to audit as code reviewers to make sure we aren't going out-of-bounds of the ptr.
```suggestion
  if (az_span_sizeprexisting_query_parameter_value) != 0)
```

Reply:
can't.  a query parameter can be 0 length.  Like `?query=&query2=&query3=value`
In that case, the az_span value we return is length 0 but with a valid ptr.

The only way we know they query was not found is with ptr=NULL

Reply:
Ah, I see. Do you think it is feasible for us to change `_az_http_request_find_query_parameter` to return `az_result` and `out az_span` and maybe have the `az_result` indicate whether we found it or not (if we already have an enum value that would work in this case)?

That will help reduce dependency on checks like these which differentiate between empty span and a span with null ptr.